File: CS/CS.NET/Windows Forms .NET/FormToForm/EventArgs/CS2008FormToForm/CS2008FormToForm/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;


public partial class Form2 : Form
{
    public Form2()
    {
        InitializeComponent();
    }

    internal event CustomEventHandler Form2TextBoxEvent;

    internal void Form1_Form1TextBoxEvent(object sender, CustomEventArgs e)
    {
        textBox1.Text = e.TextBoxTextChanged;
    }

    private void OnForm2TextBoxEvent(CustomEventArgs e)
    {
        if(Form2TextBoxEvent != null)
            Form2TextBoxEvent(this, e);
    }

    private void textBox1_TextChanged(object sender, EventArgs e)
    {
        CustomEventArgs Form2TextBoxEventArgs = new CustomEventArgs(textBox1.Text);
        OnForm2TextBoxEvent(Form2TextBoxEventArgs);
    }

    private void textBox1_KeyPress(object sender, KeyPressEventArgs e)
    {
        if (((int)e.KeyChar) == 13)
            this.Close();
    }
}

File: CS/CS.NET-VB.NET/FormToForm/VB2008FormToForm/VB2008FormToForm/Form2.vb
Public Class Form2

    ' internal event CustomEventHandler Form2TextBoxEvent;
    Friend Event Form2TextBoxEvent As CustomEventHandler

    Friend Sub Form1_Form1TextBoxEvent(ByVal sender As System.Object, ByVal e As CustomEventArgs)

        TextBox1.Text = e.TextBoxTextChanged

    End Sub

    Private Sub OnForm2TextBoxEvent(ByVal e As CustomEventArgs)

        Try
            RaiseEvent Form2TextBoxEvent(Me, e)
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
        End Try

    End Sub

    Private Sub TextBox1_TextChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles TextBox1.TextChanged

        Dim Form2TextBoxEventArgs As New CustomEventArgs(TextBox1.Text)
        OnForm2TextBoxEvent(Form2TextBoxEventArgs)

    End Sub

    Private Sub TextBox1_KeyPress(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles TextBox1.KeyPress

        If (Asc(e.KeyChar) = 13) Then
            Me.Close()
        End If

    End Sub

End Class